controllers/nnf_workflow_controller.go

```diff
@@ -1484,155 +1466,69 @@ func (r *NnfWorkflowReconciler) handlePostRunState(ctx context.Context, workflow
 	return ctrl.Result{}, nil
 }
 
-func (r *NnfWorkflowReconciler) handleTeardownState(ctx context.Context, workflow *dwsv1alpha1.Workflow, driverID string, log logr.Logger) (ctrl.Result, error) {
-	log.Info(workflow.Status.State)
-
-	exists, err := r.deleteNnfAccesses(ctx, workflow, driverID, log)
+func (r *NnfWorkflowReconciler) handleTeardown(ctx context.Context, workflow *dwsv1alpha1.Workflow, log logr.Logger) (bool, error) {
+	// Add owner labels to the persistentStorageInstances named by a "delete_persistent" #DW.
+	// They'll get deleted below in the call to deleteChildren
+	err := r.teardownPersistentStorageInstances(ctx, workflow)
 	if err != nil {
-		return ctrl.Result{}, err
+		return true, err
 	}
 
-	if exists == resourceExists {
-		return ctrl.Result{}, nil
+	childObjects := []dwsv1alpha1.ObjectList{
+		&nnfv1alpha1.NnfDataMovementList{},
+		&nnfv1alpha1.NnfAccessList{},
+		&nnfv1alpha1.NnfStorageList{},
```

Comment:
The NnfStorageProfile goes here as well.  --Actually, the nnfstorage owns the nnfstorageprofile.